skycoin/skywire
# Feature/skychat: basic p2p and group chat implementation

Did you run `make format && make check`?

Yes 

Fixes #	

 Changes:	
- Add basic support of p2p and group chats.
- Add possibility to set user name and info

How to test this PR:
1. build and install on two visors
2. open both skychat UIs
3. press on user icon to set user name and user info
4. press on "Add local server and rooms" and type in at least an alias to generate a new server with a default room
5. on skychat 'B' press on Add Remote Server and Rooms and type in all public keys you see on the user 'A's group info to connect to server

Known Bugs:
<del>1. Sometimes the transports seem to hang up and only one can send to the other but not back</del>
<del>2. the sse connection of the browsers (to notify the browser that he has to do another http request to update the visor) doesnt work properly, so after sending and receiving messages one has to refresh the browser about 80% of the times.</del>
<del> 3. Sending info with a picture does not work as it seems the picture needs more than one packet to be sent, so something needs to be implemented to send long data over skywire and put it back together </del>

Preview:
![Unbenannt](https://user-images.githubusercontent.com/50741854/219949255-949aa5bc-6d5a-4cf1-a363-78eb7d511b63.png)
